pandas/tests/test_timedelta64_overflow.py

```diff
@@ -0,0 +1,322 @@
+"""
+Check overflow behavior of operations on timedelta-valued
+DataFrames/Series/Indexes/ExtensionArrays.
+"""
+
+from itertools import (
+    chain,
+    combinations_with_replacement,
+    product,
+)
+from operator import attrgetter
+from typing import (
+    NamedTuple,
+    Type,
+    Union,
+)
+
+from hypothesis import given
+import hypothesis.strategies as st
+import pytest
+
+from pandas.errors import OutOfBoundsDatetime
+
+from pandas import (
+    DataFrame,
+    DatetimeIndex,
+    Index,
+    NaT,
+    Series,
+    Timedelta,
+    TimedeltaIndex,
+    Timestamp,
+    array,
+    to_timedelta,
+)
+import pandas._testing as tm
+from pandas.core.arrays import (
+    DatetimeArray,
+    ExtensionArray,
+    TimedeltaArray,
+)
+
+timedelta_types = (Timedelta, TimedeltaArray, TimedeltaIndex, Series, DataFrame)
+timestamp_types = (Timestamp, DatetimeArray, DatetimeIndex, Series, DataFrame)
+containers = slice(1, None)
+get_item_names = lambda t: "-".join(map(attrgetter("__name__"), t))
+
+
+# TODO: consolidate remaining td64 overflow tests here?
+#   - tests/tslibs/test_conversion.py::test_ensure_timedelta64ns_overflows()
+#   - tests/tslibs/test_timedeltas.py::test_huge_nanoseconds_overflow()
+#   - tests/scalar/timedelta/test_timedelta.py::test_mul_preserves_reso()
+#   - tests/scalar/timedelta/test_constructors.py::test_construct_from_td64_with_unit(),
+#     test_overflow_on_construction()
+
+
+class BinaryOpTypes(NamedTuple):
+    """
+    The expected operand and result types for a binary operation.
+    """
+
+    left: Type
+    right: Type
+    result: Type
+
+    def __str__(self) -> str:
+        return get_item_names(self)
+
+    def __repr__(self) -> str:
+        return f"BinaryOpTypes({self})"
+
+
+positive_tds = st.integers(min_value=1, max_value=Timedelta.max.value).map(Timedelta)
+
+xfail_no_overflow_check = pytest.mark.xfail(reason="No overflow check")
+
+
+@given(
+    st.integers(
+        min_value=Timedelta.max.value - 2**9 + 1,
+        max_value=Timedelta.max.value,
+    ).map(Timedelta)
+)
+def test_td64_summation_raises_spurious_overflow_error_for_single_elem_series(
+    value: Timedelta,
+):
+    s = Series(value)
+
+    msg = "int too big to convert|Python int too large to convert to C long"
```

Comment:
nitpick: can you write these as `msg = "|".join(["foo", "bar"])` instead of `msg = "foo|bar"`